admin/admin.go

```diff
@@ -16,6 +17,11 @@ type Options struct {
 	ProvisionerSpec string
 }
 
+type updateTSCache struct {
+	cache map[string]bool
+	lock  sync.Mutex
+}
```

Comment:
Some ways to make this cleaner:

- Rename to `usedFlusher` and move to a separate file `used_flusher.go`
- Rename `cache` to `deployments` (in the future, we may use this same object to track `used_on` for other objects, like access tokens)
- Add the flushing related methods on this struct instead of the `Service`
- Add a `newUsedFlusher(logger, db)` method that initializes it and starts the background flushing
- Add a `.close() error` method that stops the background flushing. Call it from `Service.Close`